crates/sui-types/src/messages.rs

```diff
@@ -540,32 +544,36 @@ impl<S> TransactionEnvelope<S> {
             return Ok(());
         }
 
-        let (signature, public_key) = self
-            .tx_signature
-            .get_verification_inputs(self.data.sender)?;
-        let key = public_key
-            .try_into()
-            .map_err(|_| SuiError::InvalidSignature {
-                error: "Invalid public key".to_owned(),
-            })?;
-
-        obligation
-            .public_keys
-            .get_mut(idx)
-            .ok_or(SuiError::InvalidAuthenticator)?
-            .push(key);
-        obligation
-            .signatures
-            .get_mut(idx)
-            .ok_or(SuiError::InvalidAuthenticator)?
-            .add_signature(signature)
-            .map_err(|_| SuiError::InvalidSignature {
-                error: "Failed to add signature to obligation".to_string(),
-            })?;
+        // let (signature, public_key) = self
+        //     .tx_signature
+        //     .get_verification_inputs(self.data.sender)?;
+        // let key = public_key
+        //     .try_into()
+        //     .map_err(|_| SuiError::InvalidSignature {
+        //         error: "Invalid public key".to_owned(),
+        //     })?;
+
+        // obligation
+        //     .public_keys
+        //     .get_mut(idx)
+        //     .ok_or(SuiError::InvalidAuthenticator)?
+        //     .push(key);
+        // obligation
+        //     .signatures
+        //     .get_mut(idx)
+        //     .ok_or(SuiError::InvalidAuthenticator)?
+        //     .add_signature(signature)
+        //     .map_err(|_| SuiError::InvalidSignature {
+        //         error: "Failed to add signature to obligation".to_string(),
+        //     })?;
 
         Ok(())
```

Comment:
if it makes sense for us to refactor this flow, I'd do it in a second PR (and checking we have comprehensive tests for the unhappy path) rather than aggressively switching everything to immediate verification.

Reply:
Added the ability to statically check if user's signature scheme matches the authority's signature scheme and adds it for batch verification if so - otherwise, independently verify. This should preserve the load. I was thinking about the multiple accumulators, and I suppose I just haven't imagined a great way to implement it yet, and as we still don't need it - shall we attempt at that in another PR?